Formula/kubecm.rb

```diff
@@ -0,0 +1,27 @@
+class Kubecm < Formula
+  desc "KubeConfig Manager"
+  homepage "https://guoxudong.io/post/kubecm/"
+  url "https://github.com/sunny0826/kubecm/archive/v0.10.2.tar.gz"
+  sha256 "1c5527a783c15f3389d0b8505a028704fd7238492d6ae511831cdaab32626c2f"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build",
+           "-ldflags", "-X github.com/sunny0826/kubecm/cmd.kubecmVersion=#{version}",
+           "-o", bin/"kubecm"
+
+    # Install bash completion
+    output = Utils.safe_popen_read("#{bin}/kubecm", "completion", "bash")
+    (bash_completion/"kubecm").write output
+
+    # Install zsh completion
+    output = Utils.safe_popen_read("#{bin}/kubecm", "completion", "zsh")
+    (zsh_completion/"_kubecm").write output
+  end
+
+  test do
+    assert_match "\nKubeConfig Manager\n", shell_output("#{bin}/kubecm -h 2>&1")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
fixed